tools/flang1/flang1exe/semfunc.c

```diff
@@ -5817,6 +5817,21 @@ ref_pd(SST *stktop, ITEM *list)
       XFR_ARGAST(1);
     }
     break;
+  case PD_isnan:
+    if (count != 1 || get_kwd_args(list, count, KWDARGSTR(pdsym)))
+      goto bad_args;
```

Comment:
bad_args (here and also below) does not seem to be erroring out if provided incorrect parameters. This usually results in a linking error. 
Using E74_CNT, E74_ARG might be more appropriate here. Also add a test which checks these.